apartmenttherapy/oxygen-lib-ui
# Shannon/ad no timer out of viewport

#### What does this PR do & why?
- checks if window is not in focus and stops timers if so
- checks if window is in focus and restarts timers if so

#### How can this be manually tested?
- navigate to any post page (http://localhost:8080/whole30-approved-foods-whole-foods-260963?gentest=true)
- Right click on the ad in the right rail, wait 30 seconds and watch the `data-google-query-id` update
- click on a new tab or open a new window and see the ad no longer update

For faster testing:
- go to components > Ad > index.js line 54
- replace `this.props.refreshInterval` with 5
- see ad slot update every 5 second

- see that all tests for the Ad component pass


#### Any questions or concerns about this change?

_(Any significant changes you're introducing? What side effects might this have? What other parts of the app could this affect?)_

#### Deployment Notes

_(e.g. Does this require a migration or special infrastructure change? Clearing the cache? Third party service?)_

#### Link to story card and other documentation
https://app.asana.com/0/543787180217047/759556625512875

#### GIF

#### Have you remembered to...?
